protocol-units/bridge/sources/METHBridge.move

```diff
@@ -0,0 +1,455 @@
+
+        // Transfer the coins back to the owner
+        let PendingWithdrawalRequest { owner, token_id: _, amount } = request;
+        let coin = coin::withdraw<AptosCoin>(trusted, amount);
+        coin::deposit(owner, coin);
+
+        // Emit an event to signal that the withdrawal request was closed
+        event::emit_event(
+            &mut bridge_account.pending_withdrawal_events,
+            PendingWithdrawalEvent {
+                nonce,
+                owner,
+                token_id,
+                amount: 0,
+            }
+            ,
+        );
+    }
+
+    public entry fun claim_withdrawal_request(
+        owner: &signer,
+        owner_address: address,
+        token_id: u128,
+        nonce: u256
+    ) acquires BridgeAccount {
+        // Verify that the caller of this function owns the address they claim to
+        assert!(owner_address == signer::address_of(owner), 1); // 1 indicates an authorization error
+
+        let bridge_account = borrow_global_mut<BridgeAccount>(BRIDGE_ACCOUNT);
+    
+        // Find the index of the pending withdrawal that matches the provided parameters
+        let pending_withdrawal_index = find_pending_withdrawal_index(
+            bridge_account, 
+            owner_address, 
+            token_id, 
+            nonce
+        );
+        assert!(pending_withdrawal_index != vector::length(&bridge_account.pending_withdrawals), EPENDING_WITHDRAWAL_NOT_FOUND); // 2 indicates that the pending withdrawal was not found
+    
+        // Remove the pending withdrawal from the list
+        let PendingWithdrawal { request: _, nonce: _ } = vector::remove(
+            &mut bridge_account.pending_withdrawals, 
+            pending_withdrawal_index
+        );
+    
+        // Emit an event to signal that the withdrawal claim was processed
+        event::emit_event(
+            &mut bridge_account.pending_withdrawal_events, 
+            PendingWithdrawalEvent {
+                nonce,
+                owner: owner_address,
+                token_id,
+                amount: 0,
+            }
+        );
+    }
+
+    fun find_pending_withdrawal_index(
+        bridge_account: &BridgeAccount,
+        owner: address,
+        token_id: u128,
+        nonce: u256,
+    ): u64 {
+        let pending_withdrawals = &bridge_account.pending_withdrawals;
+        let i = 0;
+        let len = vector::length(pending_withdrawals);
+        while(i < len) {
+            let pending_withdrawal = vector::borrow(pending_withdrawals, i);
+            if (pending_withdrawal.request.owner == owner &&
+                pending_withdrawal.request.token_id == token_id &&
+                pending_withdrawal.nonce == nonce) {
+                return i
+            };
+            i = i + 1;
+        };
+        len
+    }
+
+    fun init_module(bridge: &signer) {
+        let bridge_address = signer::address_of(bridge);
```

Comment:
We might need to think how we are managing this bridge address.  It seems here we could just store the bridge account and then potentially have some god function to rotate it if needed/compromised